arch/x86/interrupt.c

```diff
@@ -266,16 +266,21 @@ int interrupt_window_vmexit_handler(struct vcpu *vcpu)
 
 int external_interrupt_vmexit_handler(struct vcpu *vcpu)
 {
-	int vector = exec_vmread(VMX_EXIT_INT_INFO) & 0xFF;
+	uint32_t intr_info;
 	struct intr_ctx ctx;
 
-	ctx.vector = vector;
+	intr_info = exec_vmread(VMX_EXIT_INT_INFO);
+	if ((intr_info & VMX_INT_INFO_VALID) &&
+		((intr_info & VMX_INT_TYPE_MASK) == VMX_INT_TYPE_EXT_INT)) {
 
-	dispatch_interrupt(&ctx);
+		ctx.vector = intr_info & 0xFF;
+		dispatch_interrupt(&ctx);
 
-	VCPU_RETAIN_RIP(vcpu);
+		TRACE_2L(TRC_VMEXIT_EXTERNAL_INTERRUPT,
+					(intr_info & 0xFF), 0);
+	}
```

Comment:
I think we changed the external interrupt delivery mechanism from "acknowledge interrupt on exit" = 1 to 0. With this change, we don't need to do anything to external_interrupt_vmexit_handler.

I remember Jason ever has a patch for this. It may be missed.. @JasonChenCJ

Reply:
@dongyaozu  , Thanks for review, have remove the changes in 
 'external_interrupt_vmexit_handler()',